packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx

```diff
@@ -1,24 +1,286 @@
+            user: {
+                isSetupComplete: false,
+                organizationName: '',
+            },
+            health: {
+                mode: LightdashMode.DEFAULT,
+            },
+        });
+
+        const welcomeModel = await screen.findByRole('dialog', {
+            name: 'Nearly there... Tell us a bit more about yourself',
+        });
+        expect(welcomeModel).toBeInTheDocument();
+
+        const anonymizeTrackingCheckbox = await screen.findByRole('checkbox', {
+            name: `Anonymize my usage data`,
+        });
+        expect(anonymizeTrackingCheckbox).toBeInTheDocument();
+    });
+
+    it("should submit user's completion with correct data", async () => {
+        const user = userEvent.setup();
+
+        renderWithProviders(<UserCompletionModal />, {
+            user: {
+                isSetupComplete: false,
+                organizationName: '',
+                email: 'demo@lightdash.com',
+            },
+            health: {
+                mode: LightdashMode.DEFAULT,
+            },
+        });
+
+        // submit button should be disabled until organization name and role are filled
+        const submitButton = await screen.findByRole('button', {
+            name: 'Next',
+        });
+        expect(submitButton).toBeDisabled();
+
+        // fill in organization name
+        const nameInput = await screen.findByPlaceholderText(
+            'Enter company name',
+        );
+        expect(nameInput).toBeInTheDocument();
+        await user.type(nameInput, 'test organization');
+
+        // select role
+        const roleSelect = await screen.findByPlaceholderText(
+            'Select your role',
+        );
+        expect(roleSelect).toBeInTheDocument();
+        await user.click(roleSelect);
+        const roleOption = await screen.findByText('Software Engineer');
+        await user.click(roleOption);
+
+        // uncheck email domain checkbox (checked by default)
+        const emailDomainCheckbox = await screen.findByRole('checkbox', {
+            name: `Allow users with @lightdash.com to join the organization as a viewer`,
+        });
+        expect(emailDomainCheckbox).toBeInTheDocument();
+        expect(emailDomainCheckbox).toBeChecked();
+        await user.click(emailDomainCheckbox);
+
+        // uncheck subscription checkbox (unchecked by default)
+        const subscriptionCheckbox = await screen.findByRole('checkbox', {
+            name: `Keep me updated on new Lightdash features`,
+        });
+        expect(subscriptionCheckbox).toBeInTheDocument();
+        expect(subscriptionCheckbox).toBeChecked();
+        await user.click(subscriptionCheckbox);
+
+        // check tracking checkbox (unchecked by default)
+        const trackingCheckbox = await screen.findByRole('checkbox', {
+            name: `Anonymize my usage data`,
+        });
+        expect(trackingCheckbox).not.toBeChecked();
+        expect(trackingCheckbox).toBeInTheDocument();
```

Comment:
guess this isn't needed if 205 is true no?

Reply:
or maybe just swap?

Reply:
swap it is!